proxy/src/main/java/com/velocitypowered/proxy/connection/registry/DimensionData.java

```diff
@@ -0,0 +1,136 @@
+package com.velocitypowered.proxy.connection.registry;
+
+import com.google.common.base.Optional;
+import com.google.common.base.Preconditions;
+import net.kyori.nbt.CompoundTag;
+
+public final class DimensionData {
+  private final String registryIdentifier;
+  private final boolean isNatural;
+  private final float ambientLight;
+  private final boolean isShrunk;
+  private final boolean isUltrawarm;
+  private final boolean hasCeiling;
+  private final boolean hasSkylight;
+  private final Optional<Long> fixedTime;
+  private final Optional<Boolean> hasEnderdragonFight;
+
+  /**
+   * Initializes a new {@link DimensionData} instance.
+   * @param registryIdentifier the identifier for the dimension from the registry.
+   * @param isNatural indicates if the dimension use natural world generation (e.g. overworld)
+   * @param ambientLight the light level the client sees without external lighting
+   * @param isShrunk indicates if the world is shrunk, aka not the full 256 blocks (e.g. nether)
+   * @param isUltrawarm internal dimension warmth flag
+   * @param hasCeiling indicates if the dimension has a ceiling layer
+   * @param hasSkylight indicates if the dimension should display the sun
+   * @param fixedTime optional. If set to any game daytime value will deactivate time cycle
+   * @param hasEnderdragonFight optional. Internal flag used in the end dimension
+   */
+  public DimensionData(String registryIdentifier, boolean isNatural,
+                       float ambientLight, boolean isShrunk, boolean isUltrawarm,
+                       boolean hasCeiling, boolean hasSkylight,
+                       Optional<Long> fixedTime, Optional<Boolean> hasEnderdragonFight) {
```

Comment:
It's not a good practice to pass `Optional` as a parameter. Mark these as `@Nullable` and handle the `null` situation gracefully.

Reply:
Will revert that- Was an idea anyway